src/omv/img/imlib.c

```diff
@@ -594,20 +594,21 @@ void imlib_image_operation(image_t *img, const char *path, image_t *other, int s
         // window is equal to an image in size. However, since this is not the
         // case we shrink the window size to how many lines we're buffering.
         temp.pixels = alloc;
-        temp.h = (size / (temp.w * temp.bpp)); // round down
+        // Set the max buffer height to image height.
+        temp.h = IM_MIN(img->h, (size / (temp.w * temp.bpp)));
         // This should never happen unless someone forgot to free.
         if ((!temp.pixels) || (!temp.h)) {
             nlr_raise(mp_obj_new_exception_msg(&mp_type_MemoryError,
                                                "Not enough memory available!"));
         }
         for (int i=0; i<img->h; i+=temp.h) { // goes past end
-            int can_do = IM_MIN(temp.h, img->h-i);
-            imlib_read_pixels(&fp, &temp, 0, can_do, &rs);
```

Comment:
@kwagyeman Wouldn't this start reading from zero every time ?

Reply:
Um, I don't have the context for this code stored in my head. But, it's just reading a line at a time. You may be breaking the code with this commit. Test it. I don't believe read pixels resets itself. Line start and line end are just how many lines to read. So, the 0 doesn't really mean anything.

Reply:
I see now, line_end is actually the number of lines to read not a line offset (it should be called line_count). This works because you're reading from files and the file pointer is where you last stopped. I think line_start is misleading and should be removed.

Reply:
Okay I reverted this change to 0, but line_start should be removed and line_end renamed to line_count

Reply:
Okay, so, basically, the error was that the temp.h buffer could get too big.